Library/Homebrew/livecheck/strategy/github_latest.rb

```diff
@@ -73,32 +71,80 @@ def self.generate_input_values(url)
           match = url.sub(/\.git$/i, "").match(URL_MATCH_REGEX)
           return values if match.blank?
 
-          # Example URL: `https://github.com/example/example/releases/latest`
-          values[:url] = "https://github.com/#{match[:username]}/#{match[:repository]}/releases/latest"
+          values[:url] = "https://api.github.com/repos/#{match[:username]}/#{match[:repository]}/releases/latest"
+          values[:username] = match[:username]
+          values[:repository] = match[:repository]
 
           values
         end
 
-        # Generates a URL and regex (if one isn't provided) and passes them
-        # to {PageMatch.find_versions} to identify versions in the content.
+        # Uses a regex to match the version from release JSON or, if a block is
+        # provided, passes the JSON to the block to handle matching. With
+        # either approach, an array of unique matches is returned.
+        #
+        # @param content [Array] list of releases
+        # @param regex [Regexp] a regex used for matching versions in the content
+        # @param block [Proc, nil] a block to match the content
+        # @return [Array]
+        sig {
+          params(
+            content: T::Array[T::Hash[String, T.untyped]],
+            regex:   Regexp,
+            block:   T.nilable(Proc),
+          ).returns(T::Array[String])
+        }
+        def self.versions_from_content(content, regex, &block)
+          if block.present?
+            block_return_value = if regex.present?
+              yield(content, regex)
+            else
+              yield(content)
+            end
+            return Strategy.handle_block_return(block_return_value)
+          end
+
+          content.reject(&:blank?).map do |release|
+            next if release["draft"] || release["prerelease"]
+
+            value = T.let(nil, T.untyped)
+            VERSION_KEYS.find do |key|
+              value = release[key]&.match(regex)&.to_s
+            end
+            value
+          end.compact.uniq
```

Comment:
```suggestion
          match = content["tag_name"]&.match(regex)
          return [] if match.blank?

          [match[1]]
```

The existing `release[key]&.match(regex)&.to_s` code uses the full matched text as the version instead of the first capture group. When it comes to the default regex, this means that we get a version like `v1.2.3` instead of the expected `1.2.3`. This isn't apparent for tags like `1.2.3`, so you may have been testing this on formulae/casks where the issue isn't apparent. [This is part of the reason why manual testing over a few formulae/casks isn't sufficient and I put so much effort into exhaustively testing strategy changes across all affected formulae/casks.]

This suggestion reworks the method to only match against the `tag_name` while also ensuring that we're returning the version text from the capture group.